guides/puppet-web-apps/authorization/README.md

```diff
@@ -0,0 +1,227 @@
+
+Whenever a user will perform an action (open the page, click a button) a Permission will be checked to verify if he is allowed to do it. Grouping Permissions together ("show list of users or companies") is discouraged, because it's easy to group them later in the configuration (AuthorizationRules).
+
+Let's define some example database classes that will be useful in this guide:
+
+```c#
+[Database]
+public class Invoice : Something
+{
+public bool IsSettled;
+}
+
+[Database]
+public class InvoiceRow : Something
+{
+public string Product;
+public decimal Price;
+}
+
+[Database]
+public class InvoiceInvoiceRow : Relation
+{
+[SynonymousTo(nameof(From))]
+public Invoice Invoice;
+[SynonymousTo(nameof(To))]
+public InvoiceRow InvoiceRow;
+}
+```
+
+And now, let's define permissions that we'll use to describe the authorization rules later:
+
+```
+public class ListInvoices: Permission
+{
+}
+
+public class DisplayInvoice: Permission
+{
+public Invoice Invoice {get; private set;}
+public DisplayInvoice(Invoice invoice)
+{
+this.Invoice = invoice;
+}
+}
+```
+
+Permissions can be anything as long as they derive from `Permission`, but some parts of this library work best if their constructors accept only serializables and/or database types.
+
+### Define authorization rules
+
+Now let's define rules that will decide weather the user will be granted the permissions or not.
+
+```c#
+var rules = new AuthorizationRules();
+rules.AddRule(new ClaimRule<ListInvoices, SystemUserClaim>((claim, permission) => claim.SystemUser.Name == "admin"));
+```
+
+Rules implement `IAuthorizationRule<TPermission>` interface, which defines one method:
+
+```c#
+public bool Evaluate(
+IEnumerable<Claim> claims, // each Claim represents a fact about the current user. Most popular ones are
+// SystemUserClaim and PersonClaim
+IAuthorizationEnforcement authorizationEnforcement, // this can be used to check if the current user has other permissions
+TPermission permission) // the actual permission to check
+```
+
+`SystemUserClaim` checks weather a claim of specified type (SystemUserClaim) exists and matches the provided predicate. Here, we effectively check if current user has a name "admin".
+
+Claims are a way obtain information about the current user without direct dependency on authentication system. There are two built in claims:
+* SystemUserClaim - if present it means that a current user is signed in as a specific SystemUser
+* PersonClaim - if present it means that a current user represents specific person. It is preferred mean of establishing user's identity, since it's usually more convenient to manage People than Users
+
+There are other built-in rules in `Starcounter.Authorization.Core.Rules` namespace. Feel free to use them or create your own.
```

Comment:
WDYT about putting a list of those rules here? With short description to give an overview what a developer can expect.